Tests/AutomergeTests/TestHistory.swift

```diff
@@ -2,28 +2,43 @@ import Automerge
 import XCTest
 
 class HistoryTests: XCTestCase {
-    func testChangeCountsInHeads() {
+    func testHeadsDuringChanges() throws {
         let doc = Document()
-        try! doc.put(obj: ObjId.ROOT, key: "key", value: .String("one"))
-        var heads = doc.heads()
-        XCTAssertEqual(heads.count, 1)
+        try doc.put(obj: ObjId.ROOT, key: "key", value: .String("one"))
+        XCTAssertEqual(doc.heads().count, 1)
 
-        try! doc.put(obj: ObjId.ROOT, key: "key", value: .String("two"))
-        heads = doc.heads()
-        XCTAssertEqual(heads.count, 2)
+        try doc.put(obj: ObjId.ROOT, key: "key", value: .String("two"))
+        XCTAssertEqual(doc.heads().count, 1)
+        
+        let replicaDoc = doc.fork()
+        XCTAssertEqual(doc.heads().count, 1)
+        XCTAssertEqual(replicaDoc.heads().count, 1)
+        XCTAssertEqual(doc.heads(), replicaDoc.heads())
+        
+        try replicaDoc.put(obj: ObjId.ROOT, key: "key", value: .String("three"))
+        XCTAssertEqual(doc.heads().count, 1)
+        XCTAssertEqual(replicaDoc.heads().count, 1)
+        XCTAssertNotEqual(doc.heads(), replicaDoc.heads())
+        
+        try doc.merge(other: replicaDoc)
+        XCTAssertEqual(doc.heads().count, 1)
```

Comment:
The reason this is 1 rather than 2 is that there were no _concurrent_ changes to `doc`. In git terms, the merge of `replicaDoc` into `doc` is a fast forward merge. If, after the fork but before the merge, you were to do `doc.put(...)` we _should_ see 2 heads.

Reply:
AHHH!  Took a while to sink in to my skull, but that makes significantly more sense. With changes() in place and seeing the outputs, I think it's doing what I'm expecting/wanting (showing all changes to the document).